addon/validations/factory.js

```diff
@@ -564,7 +563,7 @@ function getCPDependentKeysFor(attribute, model, validations) {
   let dependentKeys = validations.map((validation) => {
     let { options } = validation;
     let type = validation._type;
-    let Validator = type === 'function' ? BaseValidator : lookupValidator(owner, type);
+    let Validator = type === 'function' ? BaseValidator : lookupValidator(owner, type).class;
```

Comment:
Yes, the proxy does not proxy to the factory.  I linked above pointing to the RFC which explains that is by design.

https://github.com/emberjs/ember.js/blob/c05d241fee2b24de83d67da86bdbc5c8177d4266/packages/container/lib/container.js#L185

And the RFC (first bullet)
This will not work as the proxy throws (by design) when accessing anything other than class and create: https://github.com/emberjs/rfcs/blob/master/text/0150-factory-for.md#detailed-design (See first bullet)